src/core/queue.ts

```diff
@@ -172,6 +172,41 @@ export class Queue extends EventEmitter {
     );
   }
 
+  /**
+   * delByFunction
+   * @param q - queue to look in
+   * @param func - function name to delete any jobs with
+   * @param start - optional place to start looking in list (default: beginning of list)
+   * @param stop - optional place to end looking in list (default: end of list)
+   * @returns number of jobs deleted from queue
+   */
+  async delByFunction(
+    q: string,
+    func: string,
+    start: number = 0,
+    stop: number = -1
+  ) {
+    const jobs = await this.connection.redis.lrange(
+      this.connection.key("queue", q),
+      start,
+      stop
+    );
+    let numJobsDeleted: number = 0;
+    for (let i = 0; i < jobs.length; i++) {
+      const jobEncoded = jobs[i];
+      const { class: jobFunc } = JSON.parse(jobEncoded);
+      if (jobFunc === func) {
+        await this.connection.redis.lrem(
+          this.connection.key("queue", q),
+          0,
+          jobEncoded
+        );
+        numJobsDeleted++;
+      }
+    }
+    return numJobsDeleted;
```

Comment:
I'm a little worried here about the delay between reading and deleting.  If you find your job in position 4 in the range, and then delete the job in position it might have been moved to a new position, or out of the queue entirely but the time you call `lrem`.  

The good news is that `lrem` compares the contents of `jobEncoded`, so it won't delete the wrong job by accident.  The bad news is that you might miss new jobs that have been added as the queue is changing under you.  Think that will be a problem?

Reply:
That's a good point - unless you bring LUA into the mix, I guess we can't really guarantee that we deleted all the jobs.  Could you add a note to that point in the comments?  "best-effort" and all that...

Reply:
I'll merge this in and add the note on `master` - thanks for your contribution!